level2/iam.tf

```diff
@@ -0,0 +1,43 @@
+resource "aws_iam_policy" "s3_policy" {
+  name        = var.env_code
+  path        = "/"
+
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*",
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+resource "aws_iam_role" "main" {
+  name                = var.env_code
+  managed_policy_arns = ["arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"]
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+        }
+    ]
+}
+EOF
+}
+resource "aws_iam_instance_profile" "main" {
```

Comment:
could you please define aws_iam_role_policy_attachment resource and associate both aws_iam_role and aws_iam_policy.